### PR TITLE
fix(page): fix content layout with horizontal nav

### LINKED ISSUE #6921: Page - content should span width of viewport when there is no sidebar
https://pf6.patternfly.org/components/navigation/html-demos/horizontal-nav/

Screenshot 2024-07-24 at 8 39 43 PM

### PR DESCRIPTION
**Pull request by mcoker**
fixes #6921

[BackstopJS Report.pdf](https://github.com/user-attachments/files/16384084/BackstopJS.Report.pdf)

^ Not sure why the multiple file upload icons weren't showing up - when I load the urls directly locally, icons are there. I'm assuming they just didn't load for the screenshots, so I didn't check those screenshots in.

* Fixes sidebar area taking up space when there is no sidebar (horizontal nav demos show this)
* Fixes sidebar area taking up space when sidebar is collapsed on desktop
* Fixes grid-column(-start) for masthead__content when there is no toggle on mobile (horizontal nav demos show this). It was starting at 2, updates to start at 1 so masthead__content spans the full width of the masthead
* Adds 2 new page demos
  * Collapsed sidebar (not visible on desktop or greater than 1200px)
  * Expanded sidebar (shows up over page content below desktop/1200px)

### CODE REVIEW

**Review by srambach — APPROVED** [review]
Looks great; good idea to add the new examples. 🐒

File: src/patternfly/components/Masthead/masthead.scss [hunk 1]
```diff
@@ -50,6 +51,7 @@ $pf-v6-c-masthead--inset-map: build-spacer-map("none", "sm", "md", "lg", "xl", "
   --#{$masthead}--m-display-inline__main--GridColumn: 1; 
   --#{$masthead}--m-display-inline__content--GridColumn: 2;
   --#{$masthead}--m-display-inline__content--Order: 0;
+  --#{$masthead}--m-display-stack__main--toggle--content--GridColumn: 2;
```
Comment (thatblindgeye) [inline src/patternfly/components/Masthead/masthead.scss:54]: Is this meant to be duplicated from (new) line 39, or is it meant to be `--#{$masthead}--m-display-inline__main--toggle--content--GridColumn` (which is used on line 94, but isn't defined anywhere)

File: src/patternfly/components/Page/page.scss [hunk 2]
```diff
@@ -53,6 +53,8 @@ $pf-page-v6--height-breakpoint-map: build-breakpoint-map("base", "sm", "md", "lg
 
   // Container for the main content area (grid area)
   --#{$page}__main-container--ZIndex: var(--pf-t--global--z-index--xs);
+  --#{$page}__main-container--GridArea: main;
+  --#{$page}--masthead--main-container--GridArea: sidebar / sidebar / main / main; // no sidebar
```
Comment (thatblindgeye) [inline src/patternfly/components/Page/page.scss:57]: Nit: comment is on the wrong line looks like

File: src/patternfly/components/Masthead/masthead.scss (hunk 1 above)
Comment (mcoker) [reply src/patternfly/components/Masthead/masthead.scss:54]: Doh! Should be `--m-display-inline`.

**Review by thatblindgeye — APPROVED** [review]

**Pull request merged**